src/test/java/org/zalando/planb/provider/OIDCCreateTokenIT.java

```diff
@@ -77,6 +77,47 @@ public void createServiceUserToken() {
     }
 
     @Test
+    public void createServiceUserTokenUsingHostHeader() {
+        MultiValueMap<String, Object> requestParameters = new LinkedMultiValueMap<>();
+        requestParameters.add("grant_type", "password");
+        requestParameters.add("username", "testuser");
+        requestParameters.add("password", "test");
+        String basicAuth = Base64.getEncoder().encodeToString(("testclient" + ':' + "test").getBytes(UTF_8));
+
+        RequestEntity<MultiValueMap<String, Object>> request;
+        ResponseEntity<OIDCCreateTokenResponse> response;
+
+        // wrong Host header (not mapping to any realm)
+        request = RequestEntity
+                .post(URI.create("http://localhost:" + port + "/oauth2/access_token"))
+                .header("Authorization", "Basic " + basicAuth)
+                .header("Host", "token.servicesX.example.org")
+                .body(requestParameters);
+
+        try {
+            rest.exchange(request, OIDCCreateTokenResponse.class);
+            fail("Request with invalid Host header should have failed.");
+        } catch (HttpClientErrorException ex) {
+            assertThat(ex.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+            assertThat(ex.getResponseBodyAsString()).contains("No matching realm found for token.servicesX.example.org");
+        }
+
+        // Host header contains a valid realm name
```

Comment:
can we split this in another test method? would be more clear what the test is intended to, if it fail.